apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -372,7 +372,51 @@ func (api *OffersAPI) RemoteApplicationInfo(ctx context.Context, args params.Off
 
 // DestroyOffers removes the offers specified by the given URLs, forcing if necessary.
 func (api *OffersAPI) DestroyOffers(ctx context.Context, args params.DestroyApplicationOffers) (params.ErrorResults, error) {
-	return params.ErrorResults{}, nil
+	result := make([]params.ErrorResult, len(args.OfferURLs))
+
+	user, ok := api.authorizer.GetAuthTag().(names.UserTag)
+	if !ok {
+		return params.ErrorResults{}, apiservererrors.ErrPerm
+	}
+	models, err := api.getModelsFromOffers(ctx, user, args.OfferURLs...)
+	if err != nil {
+		return params.ErrorResults{}, errors.Capture(err)
+	}
+
+	for i, one := range args.OfferURLs {
+		err := api.destroyOneOffer(ctx, models[i], one, args.Force)
+		result[i].Error = apiservererrors.ServerError(err)
+	}
+
+	return params.ErrorResults{Results: result}, nil
+}
+
+func (api *OffersAPI) destroyOneOffer(ctx context.Context, offerModel offerModel, offerString string, force bool) error {
+	if offerModel.err != nil {
+		return offerModel.err
+	}
+
+	url, err := corecrossmodel.ParseOfferURL(offerString)
+	if err != nil {
+		return err
+	}
+
+	modelUUID := offerModel.model.UUID
+	if err := api.checkAPIUserAdmin(ctx, modelUUID); err != nil {
+		return apiservererrors.ErrPerm
+	}
+
+	offerService, err := api.offerServiceGetter(ctx, modelUUID)
+	if err != nil {
+		return err
+	}
+
+	offerUUID, err := offerService.GetOfferUUID(ctx, url)
+	if err != nil {
+		return err
+	}
+
+	return offerService.RemoveOffer(ctx, offerUUID, force)
```

Comment:
Should this be on the removal service instead?

Reply:
Removal of offers should remove all the relations. This implies it 100% should go through the removal service.